eng/build.ps1

```diff
@@ -141,18 +141,28 @@ foreach ($argument in $PSBoundParameters.Keys)
     "build"                { $arguments += " -build" }
     "buildtests"           { if ($build -eq $true) { $arguments += " /p:BuildTests=true" } else { $arguments += " -build /p:BuildTests=only" } }
     "test"                 { $arguments += " -test" }
-    "configuration"        { $arguments += " -configuration $((Get-Culture).TextInfo.ToTitleCase($($PSBoundParameters[$argument])))" }
     "runtimeConfiguration" { $arguments += " /p:RuntimeConfiguration=$((Get-Culture).TextInfo.ToTitleCase($($PSBoundParameters[$argument])))" }
     "framework"            { $arguments += " /p:BuildTargetFramework=$($PSBoundParameters[$argument].ToLowerInvariant())" }
     "os"                   { $arguments += " /p:TargetOS=$($PSBoundParameters[$argument])" }
     "allconfigurations"    { $arguments += " /p:BuildAllConfigurations=true" }
-    "arch"                 { $arch = $PSBoundParameters[$argument]; $arguments += " /p:ArchGroup=$arch /p:TargetArchitecture=$arch" }
     "properties"           { $arguments += " " + $properties }
+    # configuration and arch can be specified multiple times, so they should be no-ops here
+    "configuration"        {}
+    "arch"                 {}
     default                { $arguments += " /p:$argument=$($PSBoundParameters[$argument])" }
   }
 }
 
-$env:__DistroRid="win-$arch"
+foreach ($config in $configuration) {
+  $argumentsWithConfig = $arguments + " -configuration $((Get-Culture).TextInfo.ToTitleCase($config))";
+  foreach ($singleArch in $arch) {
+    $argumentsWithArch = $argumentsWithConfig + " /p:ArchGroup=$singleArch /p:TargetArchitecture=$singleArch"
+    $env:__DistroRid="win-$singleArch"
+    Invoke-Expression "& `"$PSScriptRoot/common/build.ps1`" $argumentsWithArch"
```

Comment:
Non-blocking nit: would it be possible, at least optionally, to somehow run these in parallel?

Reply:
I’ll check to see if Windows PowerShell supports that.

Reply:
IMO, this should be a follow-on change, if done.

Reply:
So there's ways to make it run in parallel, but I'm pretty sure that the restore process doesn't support being run in parallel so we'd likely end up with file locking issues during restore (since we'd have multiple different MSBuild processes running in parallel doing the same work).

Reply:
I'm certainly not asking for implementing it in this PR, I'm just curious. I can easily imagine it's a more complex piece of work.

Reply:
Presumably the build process is already using maximum machine parallelism, right? At least the compile step is.